### PR TITLE
Add Max and Min function to Vector2.hh

### PR DESCRIPTION
**Pull request by pxalcantara**
Solved issue #71 

Functions added to solved the issue mentioned above, there was implemented:
- Max(const Vector2<T> &_v);
- Min(const Vector2<T> &_v);
- Max();
- Min();

 These functions were based on the functions with the same name at Vector3.hh

### CODE REVIEW

File: include/ignition/math/Vector2.hh [hunk 1]
```diff
@@ -134,6 +136,42 @@ namespace ignition
...
 
+      /// \brief Set this vector's components to the maximum of itself and the
+      ///        passed in vector
+      /// \return[in] _v the maximum clamping vector
+      public: void Max(const Vector2<T> &_v)
+      {
+        if (_v[0] > this->data[0])
+          this->data[0] = _v[0];
+        if (_v[1] > this->data[1])
+          this->data[1] = _v[1];
```
Comment (mjcarroll) [inline include/ignition/math/Vector2.hh:147]: As I mentioned in this PR, I think I would prefer this approach: https://github.com/ignitionrobotics/ign-math/pull/132/files#diff-730d7bbc54298b9dd719d93868660fb2R136-R139

File: include/ignition/math/Vector2.hh [hunk 2]
```diff
@@ -134,6 +136,42 @@ namespace ignition
         return (this->data[0] * _v[0]) + (this->data[1] * _v[1]);
       }
 
+      /// \brief Set this vector's components to the maximum of itself and the
+      ///        passed in vector
+      /// \return[in] _v the maximum clamping vector
```
Comment (mjcarroll) [inline include/ignition/math/Vector2.hh:141]: This function doesn't return, this should probably be `\param`

File: include/ignition/math/Vector2.hh [hunk 3]
```diff
@@ -134,6 +136,42 @@ namespace ignition
...
+      {
+        if (_v[0] > this->data[0])
+          this->data[0] = _v[0];
+        if (_v[1] > this->data[1])
+          this->data[1] = _v[1];
+      }
+
+      /// \brief Set this vector's components to the minimum of itself and the
+      ///        passed in vector
+      /// \return[in] _v the minimum clamping vector
```
Comment (mjcarroll) [inline include/ignition/math/Vector2.hh:152]: This function doesn't return, this should probably be `\param`

File: include/ignition/math/Vector2.hh (hunk 1 above)
Comment (pxalcantara) [reply include/ignition/math/Vector2.hh:147]: sure! I'll change it :wink:

File: include/ignition/math/Vector2.hh (hunk 3 above)
Comment (pxalcantara) [reply include/ignition/math/Vector2.hh]: done!

**Comment by pxalcantara** [comment]
@mjcarroll should I add these functions to the Vector2.i file, as did in #137 ?

**Comment by mjcarroll** [comment]
> @mjcarroll should I add these functions to the Vector2.i file, as did in #137 ?

Yes please, I'll leave it up to you if you want to do it here or in a follow-up.

**Comment by chapulina** [comment]
> Yes please, I'll leave it up to you if you want to do it here or in a follow-up.

I agree with you that we should just bundle all these changes. I'll open a PR.

**Comment by pxalcantara** [comment]
> Yes please, I'll leave it up to you if you want to do it here or in a follow-up.

I'll do it here. By the way, once the [PR #123](https://github.com/ignitionrobotics/ign-math/pull/132) was merged, I need to update my branch to continue with this PR, right?

Thanks!

**Comment by pxalcantara** [comment]
@chapulina Did you deleted some comment? I received an e-mail with your comment but I can't see it here :confused:

**Comment by chapulina** [comment]
> @chapulina Did you deleted some comment?

haha yeah I did, I thought I was commenting on #139 but I had the wrong tab open. I was hoping no one had seen it :sweat_smile:

**Comment by mjcarroll** [comment]
> I need to update my branch to continue with this PR, right?

Yes, you can either merge `ign-math6` into your branch, or rebase your branch onto `ign-math6`

**Comment by mjcarroll** [comment]
@pxalcantara friendly ping

**Comment by pxalcantara** [comment]
> @pxalcantara friendly ping

Sorry @mjcarroll I've been a little busy, but I gonna finish this today :see_no_evil:

**Comment by pxalcantara** [comment]
@mjcarroll done, again, sorry for the delay! :wink:

**Comment by pxalcantara** [comment]
@mjcarroll I forgot to add the tests to **.rb** file, I gonna do this later!

**Comment by mjcarroll** [comment]
> How I run just the Vector2_TEST.rb to check if the test that I'm doing is correct?

You should be able to go to the build directory and filter it with `ctest`

```
cd build
ctest -R Vector2_TEST.rb
```

**Comment by pxalcantara** [comment]
> How I run just the Vector2_TEST.rb to check if the test that I'm doing is correct?
> 
> You should be able to go to the build directory and filter it with `ctest`
> 
> ```
> cd build
> ctest -R Vector2_TEST.rb
> ```

@luccosta Maybe worth add this information at the documentation that you're doing about the Ruby tests

**Comment by luccosta** [comment]
> @luccosta Maybe worth add this information at the documentation that you're doing about the Ruby tests

It's already there :) but without `cd build`, I will add.

**Comment by pxalcantara** [comment]
@mjcarroll I'm facing some problems doing the test at Vector2.br 

```
overview@px:ignition-math6$ ctest -R Vector2_TEST.rb -V
UpdateCTestConfiguration  from :/home/overview/citadel_ws/build/ignition-math6/DartConfiguration.tcl
Parse Config file:/home/overview/citadel_ws/build/ignition-math6/DartConfiguration.tcl
UpdateCTestConfiguration  from :/home/overview/citadel_ws/build/ignition-math6/DartConfiguration.tcl
Parse Config file:/home/overview/citadel_ws/build/ignition-math6/DartConfiguration.tcl
Test project /home/overview/citadel_ws/build/ignition-math6
Constructing a list of tests
Done constructing a list of tests
Updating test list for fixtures
Added 0 tests to meet fixture requirements
Checking test dependency graph...
Checking test dependency graph end
test 85
    Start 85: Vector2_TEST.rb

85: Test command: /usr/bin/ruby "-I/home/overview/citadel_ws/build/ignition-math6/lib" "/home/overview/citadel_ws/src/ign-math/src/Vector2_TEST.rb" "--gtest_output=xml:/home/overview/citadel_ws/build/ignition-math6/test_results/Vector2_TESTrb.xml"
85: Test timeout computed to be: 1500
85: Loaded suite Vector2_TEST
85: Started
85: ......E
85: ===============================================================================
85: Error: test_max(Vector2_TEST): NoMethodError: undefined method `Max' for #<Ignition::Math::Vector2d:0x000055c8df42ed40>
85: /home/overview/citadel_ws/src/ign-math/src/Vector2_TEST.rb:116:in `test_max'
85:      113:      vec2 = Ignition::Math::Vector2d.new(0.3, 0.5)
85:      114:      vec3 = Ignition::Math::Vector2d.new(0.4, 0.2)
85:      115:      
85:   => 116:      assert((vec1.Max() - 0.2).abs() < 1e-10,
85:      117:            "Vector3 vec1.Max should equal 0.3")
85:      118:      
85:      119:      vec1.Max(vec2)
85: ===============================================================================
85: ..
85: 
85: Finished in 0.004180697 seconds.
85: ------
85: 9 tests, 48 assertions, 0 failures, 1 errors, 0 pendings, 0 omissions, 0 notifications
85: 88.8889% passed
85: ------
85: 2152.75 tests/s, 11481.34 assertions/s
1/1 Test #85: Vector2_TEST.rb ..................***Failed    0.17 sec

0% tests passed, 1 tests failed out of 1

Total Test time (real) =   0.18 sec

The following tests FAILED:
	 85 - Vector2_TEST.rb (Failed)
Errors while running CTest
```
I've already added these functions to [Vector2.i](https://github.com/pxalcantara/ign-math/blob/b260c687ad765c62b23677ee15b73321da40bc6f/src/Vector2.i#L49). Could you help me and show what I'm doing wrong? 
Thanks!

**Comment by mjcarroll** [comment]
> Could you help me and show what I'm doing wrong?

Yep!

First a note that the three `ignition_math-ci` jobs pass because we don't do the Ruby bindings in them.

When I check out your branch, everything looks to pass for me locally:
```
85: ------
85: 9 tests, 51 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
85: 100% passed
85: ------
```

Maybe once #139 lands, we can rebase this and see how it looks in Github Actions?  There are obviously inconsistencies between our two systems if we are getting different results like this.

**Comment by pxalcantara** [comment]
So, as far as I understood, my code is correct, so I can continue to implement the test in the Ruby file, right? @mjcarroll @chapulina

**Comment by mjcarroll** [comment]
Yes, it currently works fine for me, feel free to continue @pxalcantara

**Comment by pxalcantara** [comment]
@mjcarroll @chapulina Apparently there is a problem with my system because this test continuos to break but, once it passed the CI check here, that's what mater. Sorry for the PR take so long but I think now is finished, right? Or do you wanna me to do more changes?

**Comment by mjcarroll** [comment]
Passing CI is sufficient for me!

**Pull request merged**